src/server/server_impl.go

```diff
@@ -300,15 +301,6 @@ func (server *server) handleGracefulShutdown() {
 		sig := <-sigs
 
 		logger.Infof("Ratelimit server received %v, shutting down gracefully", sig)
-		server.grpcServer.GracefulStop()
-		server.listenerMu.Lock()
-		if server.debugListener.listener != nil {
-			server.debugListener.listener.Close()
-		}
-		if server.httpServer != nil {
-			server.httpServer.Close()
-		}
-		server.listenerMu.Unlock()
-		os.Exit(0)
```

Comment:
I think we should leave the `Exit(0)` call after the `Stop()` call? Is it not needed?

Reply:
yes. it is needed